### ISSUE TITLE
Unable to suppress `clippy::arbitrary_source_item_ordering`

### ISSUE THREAD

**Issue by bzoracler** [rust-lang/rust-clippy#15590]
### Summary

There seems to be no valid location to place `#[expect(clippy::arbitrary_source_item_ordering)]` such that clippy warnings or errors aren't generated.

**Context**

I have a macro which takes an `enum` and generates an `impl` for the `enum`. There seems to be no way of tweaking the option `module-item-order-groupings` to allow sorting such that `impl`s are allowed to follow immediately after an `enum/struct/union`; it only allows grouping all `impl`s in a file, or setting `impl`s to be on the same grouping level as `enum/struct/union`s.

I could have suppressed the errors I was getting if clippy either
* Ignores macro-generated items when considering `arbitrary_source_item_ordering`, or
* Allows `#[expect(...)]` suppression at the error location
but unfortunately neither of these seem to be possible.

### Reproducer

In the following example,

```rust
#![deny(clippy::arbitrary_source_item_ordering)]

// #[expect(clippy::arbitrary_source_item_ordering)]
struct A {}

// #[expect(clippy::arbitrary_source_item_ordering)]
use std::default::Default;
```

all 3 combinations (uncomment above `struct A` only, uncomment above `use std::default::Default` only, and uncomment above both) generate both 
* `error: incorrect ordering of items (module item groupings specify another order)`
   ```rust
   use std::default::Default;
                     ^^^^^^^
   ```
* `warning: this lint expectation is unfulfilled`
   ```rust
   #[expect(clippy::arbitrary_source_item_ordering)]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   ```

### Version

```text
Reproducible on [rust playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=0fbdb5307803bc518b0fdeb2d02f67c5) with

clippy: 0.1.91 (2025-08-29 fe55364329)
nightly: 1.91.0 (2025-08-29 fe55364329579d361b1a)
edition: 2024
```

**Comment by SpriteOvO** [comment]
This is [documented](https://rust-lang.github.io/rust-clippy/master/index.html#arbitrary_source_item_ordering) expected behavior:

> Lints on a Contains basis
> 
> The lint can be disabled only on a “contains” basis, but not per element within a “container”, e.g. the lint works per-module, per-struct, per-enum, etc. but not for “don’t order this particular enum variant”.
